### ISSUE TITLE
label-has-for fails for bound id/label

### ISSUE THREAD

**Issue by sem4phor** [maranran/eslint-plugin-vue-a11y#15]
When using a dynamic for / id attribute on label / input elements the rule `label-has-for` fails.

```
error: Form label must have associated control (vue-a11y/label-has-for)

<template>
  <label :for="'_uid'">
    <input :id="_uid">
  </label>
</template>
```

Expected behaviour:
The rule passes.

**Comment by sem4phor** [comment]
same for `vue-a11y/alt-text` rule

**Comment by maranran** [comment]
lastest version ? I fail to reappear this bug @sem4phor

**Comment by sem4phor** [comment]
yes on the newest version. Did you test it with input nested in label and a bound for?

**Comment by venikman** [comment]
I got it work. The problem that looks like that rule was taken from `react` rules. You can see it in documentation for rule. Name of rule for react. `jsx-a11y/label-has-for"`.
So what I had to do it to add options for it. 
```json
"vue-a11y/label-has-for": [ 2, {
            "components": [ "label" ],
            "required": {
                "every": [ "nesting", "id" ]
            },
            "allowChildren": false
        }]
```
I changed `jsx` to `vue` and `Label` to `label` 
I will probably can open a PR to fix it.

**Comment by andresdameson** [comment]
I changed the rule made by venikman to support nesting OR id:

      "vue-a11y/label-has-for": [ 2, {
            "components": [ "label" ],
            "required": {
                "some": [ "nesting", "id" ]
            },
            "allowChildren": false
     }]

**Comment by ZebulanStanphill** [comment]
It's worth noting that `eslint-plugin-jsx-a11y`'s `label-has-for` has recently been deprecated and replaced by [`label-has-associated-control`](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/label-has-associated-control.md). This plugin should probably follow suit.
